components/ui-api-layer/internal/domain/k8s/resourcequota_resolver.go

```diff
@@ -30,3 +71,153 @@ func (r *resourceQuotaResolver) ResourceQuotasQuery(ctx context.Context, environ
+			status, err := r.checkPodsUsage(environment, ss.Spec.Selector.MatchLabels, resourcesToCheck, resourceQuotas)
+			if err != nil {
+				return gqlschema.ResourceQuotaStatus{}, err
+			}
+			if status {
+				return r.genericSetExceededStatus(ssKind, ss.Name, ss.Status.Replicas, *ss.Spec.Replicas), nil
+			}
+		}
+	}
+	return gqlschema.ResourceQuotaStatus{Exceeded: false}, nil
+}
+
+func (r *resourceQuotaResolver) checkPodsUsage(environment string, labelSelector map[string]string, resourcesToCheck []v1.ResourceName, resourceQuotas []*v1.ResourceQuota) (bool, error) {
+	pods, err := r.podLister.ListPods(environment, labelSelector)
+	if err != nil {
+		return false, errors.Wrapf(err, "while listing pods [environment: %s][labelSelector: %v]", environment, labelSelector)
+	}
+
+	if len(pods) > 0 {
+		limits := r.containersUsage(pods[0].Spec.Containers, resourcesToCheck)
+		for _, rq := range resourceQuotas {
+			for _, name := range resourcesToCheck {
+				hard, hardExists := rq.Spec.Hard[name]
+				used := rq.Status.Used[name]
+				if hardExists && hard.Value()-used.Value() < limits[name].Value() {
+					return true, nil
+				}
+			}
+		}
+	}
+	return false, nil
+}
+
+func (*resourceQuotaResolver) containersUsage(containers []v1.Container, resourcesToCheck []v1.ResourceName) map[v1.ResourceName]*resource.Quantity {
+	limits := make(map[v1.ResourceName]*resource.Quantity)
+	for _, name := range resourcesToCheck {
+		limits[name] = &resource.Quantity{}
+	}
+	for _, container := range containers {
+		for _, name := range resourcesToCheck {
+			switch name {
+			case v1.ResourceRequestsMemory:
+				limits[name].Add(*container.Resources.Requests.Memory())
+			case v1.ResourceRequestsCPU:
+				limits[name].Add(*container.Resources.Requests.Cpu())
+			case v1.ResourceLimitsMemory:
+				limits[name].Add(*container.Resources.Limits.Memory())
+			case v1.ResourceLimitsCPU:
+				limits[name].Add(*container.Resources.Limits.Cpu())
+			case v1.ResourcePods:
+			default:
+				glog.Infof("ui-api-layer doesn't support calculating the `%s` limit", name)
+			}
+		}
+	}
+	return limits
+}
+
+func (r *resourceQuotaResolver) extractMaxUnavailable(environment string, name string, replicas int32) (int32, error) {
+	deploy, err := r.deployGetter.Find(name, environment)
+	if err != nil {
+		return 0, errors.Wrapf(err, "while getting Deployment [name: %s][environment: %s]")
+	}
+	maxUnavailable := deploy.Spec.Strategy.RollingUpdate.MaxUnavailable.StrVal
+
+	if strings.Contains(maxUnavailable, "%") {
+		maxUnavailable = maxUnavailable[:len(maxUnavailable)-1]
+		percent, err := strconv.ParseInt(maxUnavailable, 10, 64)
+		if err != nil {
+			return 0, err
+		}
+		return int32(percent) * replicas / 100, nil
+	} else {
+		return deploy.Spec.Strategy.RollingUpdate.MaxUnavailable.IntVal, nil
+	}
+}
+
+func (*resourceQuotaResolver) genericSetExceededStatus(kind string, name string, status int32, replicas int32) gqlschema.ResourceQuotaStatus {
```

Comment:
you are not setting anything, just something like: exceededStatus?

Reply:
Done `exceededStatus` is shorter and better, `genericSet` was because it could be `StatefulSet` or any other `Set`.